core/types/l1_message_tx.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"math/big"
+
+	"github.com/scroll-tech/go-ethereum/common"
+)
+
+const L1MessageTxType = 0x7E
+
+// payload, RLP encoded
+type L1MessageTx struct {
+	Nonce  uint64
+	Gas    uint64          // gas limit
+	To     *common.Address `rlp:"nil"` // nil means contract creation
+	Value  *big.Int
+	Data   []byte
+	Sender *common.Address
```

Comment:
Do we need a pointer here?

```suggestion
	Sender common.Address
```

Reply:
idk, was just following the spec. why do we want to remove pointer?

Reply:
`To` always exists, no pointer is needed here.

Reply:
Updated the spec too